paper-dialog-scrollable.html

```diff
@@ -0,0 +1,130 @@
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-flex-layout/classes/iron-flex-layout.html">
+<link rel="import" href="../paper-styles/paper-styles.html">
+
+<!--
+`paper-dialog-scrollable` implements a scrolling area used in a Material Design dialog. Use this
+together with elements implementing `paper-dialog-behavior`.
+
+It shows a top divider after scrolling if it is not the first child in its parent container. It
+shows a bottom divider if it is scrollable and it is not the last child in its parent container.
+The bottom divider is hidden if it is scrolled to the bottom.
+-->
+
+<dom-module id="paper-dialog-scrollable">
+
+  <style>
+
+    :host {
+      display: block;
+      position: relative;
+    }
+
+    :host(.is-scrolled:not(:first-child))::before {
+      content: '';
+      position: absolute;
+      top: 0;
+      left: 0;
+      right: 0;
+      height: 1px;
+      background: var(--divider-color);
+    }
+
+    :host(.can-scroll:not(.scrolled-to-bottom):not(:last-child))::after {
+      content: '';
+      position: absolute;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      height: 1px;
+      background: var(--divider-color);
+    }
+
+    .scrollable {
+      padding: 0 24px;
+
+      mixin(--layout-scroll);
+
+      mixin(--paper-dialog-scrollable);
+    }
+  </style>
+
+  <template>
+    <div id="scrollable" class="scrollable">
+      <content></content>
+    </div>
+  </template>
+
+</dom-module>
+
+<script>
+
+(function() {
+
+  Polymer({
+
+    is: 'paper-dialog-scrollable',
+
+    enableCustomStyleProperties: true,
+
+    listeners: {
+      'scrollable.scroll': '_onScroll'
+    },
+
+    /**
+     * Returns the scrolling element.
+     */
+    get scrollTarget() {
+      return this.$.scrollable;
+    },
+
+    attached: function() {
+      this.classList.add('no-padding');
+      // Set itself to the overlay sizing target
+      this.parentNode.sizingTarget = this.scrollTarget;
+      // If the host is sized, fit the scrollable area to the container. Otherwise let it be
+      // its natural size.
+      if (this.offsetHeight > 0) {
+        this.$.scrollable.classList.add('fit');
+      }
+      // FIXME use resizeable
+      this.async(this._scroll, 100);
+    },
+
+    _scroll: function() {
+      if (this.scrollTarget.scrollTop > 0) {
+        this.classList.add('is-scrolled');
+      } else {
+        this.classList.remove('is-scrolled');
+      }
```

Comment:
I don't have a good solution, but I wish that there was a cleaner way toggle all of these classes.